### PR TITLE
Adds partial op support for M slices

### PR DESCRIPTION
**Pull request by MichaelQ94**
Closes #28

### CODE REVIEW

File: src/lib/core/partialcubeop.test.ts [hunk 1]
```diff
@@ -536,3 +536,358 @@ describe("3x3 half turn", () => {
 
   expectAllOtherFacesUnaffected(CUBE_FACE_3X3, partial.halfTurn);
 });
+
+const CUBE_M_LAYER_2X2 = {
```
Comment (itstang) [inline src/lib/core/partialcubeop.test.ts:540]: technical callout here, M doesn't exist for 2x2 (only exists for odd number of layers of course), are you just treating it as an L turn?

File: src/lib/core/partialcubeop.ts [hunk 2]
```diff
@@ -95,3 +95,138 @@ export function halfTurn(
...
+    face,
+    row,
+    column,
+  };
+}
+
+/**
+ * Computes the new CubeIndex to which the facelet occupying the input CubeIndex would be sent by
+ * an M slice. Only the U, F, D, and B faces are ever affected by M slices. Calling mSlice
+ * on a 3x3 cube with `layer == 2` has this effect (affected indices are highlighed using square
```
Comment (itstang) [inline src/lib/core/partialcubeop.ts:113]: column 2 isn't the M layer is it? shouldn't it be column 1?

File: src/lib/core/partialcubeop.ts [hunk 3]
```diff
@@ -95,3 +95,138 @@ export function halfTurn(
...
+ *
+ *                     {500} {501} [202]
+ *                     {510} {511} [212]
+ *                     {520} {521} [222]
+ *
+ * @param {CubeIndex} position The starting position of the facelet to be moved
+ * @param {number} layer The column on the U face which should be affected by this operation
+ * @param {number} dimension The dimension of the cube being acted on
+ */
+export function mSlice(
```
Comment (itstang) [inline src/lib/core/partialcubeop.ts:147]: why is `mSlice` a different operation compared to say the standard R, L turns?

Comment (MichaelQ94) [reply src/lib/core/partialcubeop.ts:147]: So these aren't "real" cube operations, just partial ones that are meant to be utilities to aid in implementing actual cubeops. For instance, R on a 3x3 will be implemented using a combination of partial.quarterTurn on the R face and partial.reverseMSlice on layer 2. Hopefully that answers your other questions as well.

**Pull request merged**